src/duty/attestation/attestation.service.ts

```diff
@@ -0,0 +1,126 @@
+import { BitVectorType, fromHexString } from '@chainsafe/ssz';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+
+import { ConfigService } from 'common/config';
+import { BeaconBlockAttestation, BlockInfoResponse, ConsensusProviderService } from 'common/eth-providers';
+import { bigintRange } from 'common/functions/range';
+import { PrometheusService, TrackTask } from 'common/prometheus';
+
+import { SummaryService } from '../summary';
+
+interface SlotAttestation {
+  bits: boolean[];
+  head: string;
+  target_root: string;
+  target_epoch: string;
+  source_root: string;
+  source_epoch: string;
+  slot: string;
+  committee_index: string;
+}
+
+@Injectable()
+export class AttestationService {
+  private readonly slotsInEpoch: bigint;
+  private savedCanonSlotsAttProperties = {};
+  public constructor(
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    protected readonly config: ConfigService,
+    protected readonly prometheus: PrometheusService,
+    protected readonly clClient: ConsensusProviderService,
+    protected readonly summary: SummaryService,
+  ) {
+    this.slotsInEpoch = BigInt(this.config.get('FETCH_INTERVAL_SLOTS'));
+  }
+
+  @TrackTask('check-attestation-duties')
+  public async check(epoch: bigint, stateSlot: bigint): Promise<void> {
+    const { blocksAttestations, allMissedSlots } = await this.getProcessedAttestations(epoch);
+    this.savedCanonSlotsAttProperties = {};
+    this.logger.log(`Getting attestation duties info`);
+    const committees = await this.clClient.getAttestationCommitteesInfo(stateSlot, epoch);
+    this.logger.log(`Processing attestation duty info`);
+    for (const committee of committees) {
+      for (const [block, blockAttestations] of Object.entries(blocksAttestations).filter(
+        ([b]) => BigInt(b) > BigInt(committee.slot), // Attestation cannot be included in the previous or current block
+      )) {
+        const attestations: SlotAttestation[] = blockAttestations.filter(
+          (att: any) => att.slot == committee.slot && att.committee_index == committee.index,
+        );
+        if (!attestations) continue; // try to find committee attestations in another next block
```

Comment:
I think you meant the following:

```suggestion
        if (!attestations.length) continue; // try to find committee attestations in another next block
```